baybe/surrogates/ngboost.py

```diff
@@ -70,7 +58,7 @@ def _make_target_scaler_factory() -> type[OutcomeTransform] | None:
         # Tree-like models do not require any output scaling
         return None
 
-    @batchify
+    @batchify_mean_var_prediction
```

Comment:
I'm a bit confused here: This class is an `IndependentGaussianSurrogate`. If we have such a class, we do not allow to use a `batch_quantity>1` and raise an error in this case. Hence, do we actually still need such a decorator at all? Or is this used for only evaluating batches of shape `t x 1 x d` (with `d` being the size of the search space). If so, where is `t` actually being relevant for/used by us?

Reply:
This is only for performance reasons. Even when you request one recommendation (i.e. both `batch_size=1` and `q=1`), you still need to avoid all potential candidates, which is done as a `t-batch`. From a functionality point of view, the decorator not required, but without it the t-batches would be evaluated in a for loop, which is incredibly slow. Here, I do the trick and convert them to a virtual q-batch, which is MUCH faster

Reply:
Ok, let me try to reformulate to see if I understand: You simply flatten the `t`-batch into the `q`-batch dimension to speed things up which we can do since we only have `q=1` for all of the relevant models?

However, I don't get your explanation at one point: What do you mean by "avoid all candidates"?

Reply:
Yes exactly 👍🏼  And sorry, the "avoid" should have been an "evaluate" (don't know how this happened 🙃 ). 

So what about the issue now? Resolved or still requires changes?

Reply:
all good, discussed via chat